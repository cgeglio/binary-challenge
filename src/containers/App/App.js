import React, { Component } from 'react';
import { Route, Redirect } from 'react-router-dom';
import './App.scss';
import Form from '../Form/Form';
import Nav from '../Nav/Nav';
import Loader from '../../components/Loader/Loader';
import CardContainer from '../CardContainer/CardContainer';
import { connect } from 'react-redux';
import banner from '../../images/banner.png';
import crystal from '../../images/crystal.png';


export class App extends Component {

  render() {
    return (
      <main>
        <Route exact path="/">
          {this.props.user.username ? <Redirect to='/home' /> :
          <section className='login-page'>
            <img src={banner} alt="Your Fate Lies in Your Hands" className='banner-text'/>
            <Form />
            <img src={crystal} alt="Two hands holding a crystal ball" className='crystal-ball-img'/>
          </section>
          }
        </Route>
        <Route exact path='/home'>
            <Nav />

            <CardContainer />
            <p>hereeeeee</p>
        </Route>
      </main>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user,
  cards: state.cards
})

export default connect(mapStateToProps)(App);


  // {!this.props.cards.length && <Loader />}
