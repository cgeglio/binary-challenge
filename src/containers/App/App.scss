@import "../../variables.scss";
@import "../../mixins.scss";

* {
  font-family: $secondary-font;
  margin: 0;
  padding: 0;
}

main {
  color: white;
  background-color: black;
  height: 100vh;
}

.banner-text {
  height: 200px;
  margin-top: 50px;
}

.crystal-ball-img {
  height: 500px;
  margin-top: 140px;
  position: absolute;
}

.login-page {
  @include flex();
  height: auto;
  padding: 20px;
  position: relative;
}
