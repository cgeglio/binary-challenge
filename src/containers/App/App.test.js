import React from 'react';
import { App, mapStateToProps } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  it ('should match the snapshot', () => {
    let mockUser = {username: 'Bithcuits', id: 9}
    let mockQuestion = 'Is Bithcuits a good kitty?'
    const wrapper = shallow(<App user={mockUser}  question={mockQuestion}/>);

    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should return a user object and a question', () => {
      let mockUser = {username: 'Bithcuits', id: 9}
      let mockQuestion = 'Is Bithcuits a good kitty?'
      const wrapper = shallow(<App  user={mockUser}  question={mockQuestion} />);
      const mockDispatch = jest.fn();
      const mockState = {
        user: {username: 'Bithcuits', id: 9},
        cards: [{name: 'The Magiciam', value: 7}],
        question: 'Is Bithcuits a good kitty?',
        currentReading: {cards: [{name: 'The Magiciam', value: 7}], id: 40},
        fortune: 'Keep your friends close and your enemies closer',
      }
      const expected = {
        user: {username: 'Bithcuits', id: 9},
        question: 'Is Bithcuits a good kitty?',
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    })
  });
})
