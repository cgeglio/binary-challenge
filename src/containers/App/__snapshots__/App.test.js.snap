// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main>
  <Route
    exact={true}
    path="/"
  >
    <Redirect
      to="/home"
    />
  </Route>
  <Route
    exact={true}
    path="/home"
  >
    <Redirect
      to="/reading"
    />
  </Route>
  <Route
    exact={true}
    path="/reading"
  >
    <Connect(Nav) />
    <Connect(CardContainer) />
  </Route>
  <Route
    exact={true}
    path="/saved"
  >
    <Connect(Nav) />
    <Connect(PreviewContainer) />
  </Route>
  <Route
    exact={true}
    path="/deck"
  >
    <Connect(Nav) />
    <Deck />
  </Route>
  <Route
    exact={true}
    path="/about"
  >
    <Connect(Nav) />
    <About />
  </Route>
  <Route
    exact={true}
    path="/saved/:id"
  >
    <Connect(Nav) />
    <Connect(FavoriteContainer) />
  </Route>
</main>
`;
